### PR TITLE
migrate to hardhat

### PR DESCRIPTION
**Pull request by zhouhanseng**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 15
+      - run: npm ci
+      - run: npm test
```
Comment (septs) [inline .github/workflows/test.yaml:20]:
```suggestion
      - run: npm cit
```

from https://docs.npmjs.com/cli/v7/commands/npm-install-ci-test

File: package.json [hunk 2]
```diff
@@ -1,13 +1,33 @@
 {
+    "name": "initial-twitter-offering",
+    "version": "1.0.0",
+    "license": "MIT",
+    "author": "Yisi Liu",
+    "main": "hardhat.config.ts",
+    "scripts": {
+        "test": "hardhat test",
```
Comment (septs) [inline package.json:8]: suggest, integrate husky on pre-commit, automate compile and test

Comment (zhouhanseng) [reply package.json:8]: I'd like to have a try

**Comment by yisiliu** [comment]
closes #17

**Pull request merged**